.github/workflows/deploy.yaml

```
@@ -32,5 +51,5 @@ jobs:
         id: short-sha
       - run: yarn subgraph deploy ${{ matrix.chain }}
         env:
-          TOKEN: ${{ secrets.THEGRAPH_TOKEN }}
+          TOKEN: ${{ matrix.chain == 'core' && secrets.THEGRAPH_CORE_TOKEN || secrets.THEGRAPH_TOKEN }}
```

Not related to your PR, but still important, I think with `mantle`, we should not send any token, to prevent leaking our TheGraph token. cc @MantisClone Do you think this will work?
```suggestion
          TOKEN: ${{ matrix.chain == 'mantle-testnet' && "" 
                     || matrix.chain == 'mantle' && ""
                     || matrix.chain == 'core' && secrets.THEGRAPH_CORE_TOKEN
                     || secrets.THEGRAPH_TOKEN }}
```

---

Or that would be more secure like @benjlevesque 's proposed, with a different step or job, this way we won't risk leaking it again in the future.

---

Yes, this should work. The `mantle` and `mantle-testnet` graph nodes don't require any sort of token to deploy.

Oops. I should have been more careful to not leak `THEGRAPH_TOKEN` when I implemented the mantle deployment. Good catch @alexandre-abrioux

---

@MantisClone I was wondering if there's no token required for deployment, how is the subgraph protected from anyone updating it ?

---

This is a great question which I have forwarded to the Mantle Foundation.

---

The FusionX team responded re: Mantle deployment. They said they only allow deployments during specific time windows and disallow deployments outside these time windows. I'm inquiring to learn when these time windows occur and the process for initiating a deployment.

I have not heard back from Mantle yet re: Mantle Testnet deployment.